lib/src/loaders.dart

```diff
@@ -0,0 +1,213 @@
+import 'package:vector_graphics/vector_graphics.dart';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart';
+
+/// Compute will mess up tests because of FakeAsync.
+final bool _isTest = Platform.executable.endsWith('flutter_tester') ||
+    Platform.executable.endsWith('flutter_tester.exe');
+
+final ComputeImpl _computeImpl = _isTest
+    ? <Q, R>(ComputeCallback<Q, R> callback, Q message,
+            {String? debugLabel}) async =>
+        await callback(message)
+    : compute;
+
+/// A [BytesLoader] that parses an SVG string in an isolate and creates a
+/// vector_graphics binary representation.
+class SvgStringLoader extends BytesLoader {
+  /// See class doc.
+  const SvgStringLoader(
+    this.svg, {
+    this.theme = const SvgTheme(),
+  });
+
+  /// The raw XML string.
+  final String svg;
+
+  /// The theme to determine currentColor and font sizing attributes.
+  final SvgTheme theme;
+
+  @override
+  Future<ByteData> loadBytes(BuildContext context) async {
+    return await _computeImpl((String svg) async {
+      final Uint8List compiledBytes = await encodeSvg(
+        xml: svg,
+        theme: theme,
+        debugName: 'Svg loader',
+        enableClippingOptimizer: false,
+        enableMaskingOptimizer: false,
+        enableOverdrawOptimizer: false,
+      );
+      return compiledBytes.buffer.asByteData();
+    }, svg, debugLabel: 'Load Bytes');
+  }
+}
+
+/// A [BytesLoader] that decodes and parses a UTF-8 encoded SVG string from a
+/// [Uint8List] in an isolate and creates a vector_graphics binary
+/// representation.
+class SvgBytesLoader extends BytesLoader {
+  /// See class doc.
+  const SvgBytesLoader(
+    this.svg, {
+    this.theme = const SvgTheme(),
+  });
+
+  /// The UTF-8 encoded XML bytes.
+  final Uint8List svg;
+
+  /// The theme to determine currentColor and font sizing attributes.
+  final SvgTheme theme;
+
+  @override
+  Future<ByteData> loadBytes(BuildContext context) async {
+    return await _computeImpl((_) async {
+      final Uint8List compiledBytes = await encodeSvg(
+        xml: utf8.decode(svg),
+        theme: theme,
+        debugName: 'Svg loader',
+        enableClippingOptimizer: false,
+        enableMaskingOptimizer: false,
+        enableOverdrawOptimizer: false,
+      );
+      return compiledBytes.buffer.asByteData();
+    }, null, debugLabel: 'Load Bytes');
+  }
+}
+
+/// A [BytesLoader] that decodes SVG data from a file in an isolate and creates
+/// a vector_graphics binary representation.
+class SvgFileLoader extends BytesLoader {
+  /// See class doc.
+  const SvgFileLoader(
+    this.file, {
+    this.theme = const SvgTheme(),
+  });
+
+  /// The file containing the SVG data to decode and render.
+  final File file;
+
+  /// The theme to determine currentColor and font sizing attributes.
+  final SvgTheme theme;
+
+  @override
+  Future<ByteData> loadBytes(BuildContext context) async {
+    final Uint8List bytes = file.readAsBytesSync();
```

Comment:
Since this is already async I would the async file API

Reply:
I always avoid it and I'm not entirely sure why, but I believe it's generally more expensive on the implementation side right?

Reply:
For things like exists, the sync version is much faster. But for reading the bytes you're potentially blocking the UI isolate for multiple milliseconds if disk I/O is getting used elsewhere.

Reply:
Ahhh I meant to put this in the compute isolate

Reply:
Yeah that's much better, we can avoid copying the bytes an extra time too